x-pack/plugins/encrypted_saved_objects/server/create_migration.ts

```diff
@@ -62,24 +63,30 @@ export const getCreateMigration = (
       return encryptedDoc;
     }
 
-    const descriptor = {
-      id: encryptedDoc.id!,
-      type: encryptedDoc.type,
-      namespace: encryptedDoc.namespace,
-    };
+    // If a document has been converted right before this migration function is called, it will no longer have a `namespace` field, but it
+    // will have a `namespaces` field; in that case, the first/only element in that array should be used as the namespace in the descriptor
+    // during decryption.
+    const { convertToMultiNamespaceType } = inputType ?? {};
+    const decryptDescriptorNamespace = convertToMultiNamespaceType
+      ? normalizeNamespace((encryptedDoc.namespaces ?? [])[0])
```

Comment:
The normalization is to change what I call a "namespace string" into a classic representation of a namespace.

In a nutshell, it does one thing: if the namespace string is `"default"`, it changes the value to `undefined`. Otherwise it doesn't modify the string at all.

I'll apply the suggested change (thanks!) and add a comment here to clarify.